Powder bed fusion exampled and code clean up and readability edits

tomflint22 opened the pull request.

tomflint22 committed:
added powder based manufacturing tutorial cases based on e-mail requests

philipcardiff committed:
Make aesthetic changes related to style and also compatability changes for the Clang compiler e.g. unambiguous field construction

philipcardiff committed:
Remove temporary build files

philipcardiff committed:
Add high-level build scripts

tomflint22 committed:
fixed the powder bed 2D case as the heat source was too large and the decompose rule was wrong for beamweldfoam

philipcardiff committed:
Remove .DS_Store file (macOS index)

philipcardiff committed:
Add tutorials/Allclean and remove backup text files

philipcardiff committed:
Move tutorials from applications to parent directory

philipcardiff committed:
PowderBed2D: add Allrun and Allclean scripts

philipcardiff committed:
PowderBed2D/All*: Change sh to bash

philipcardiff committed:
PowderBed2D/Allrun: use runParallel

philipcardiff committed:
Add All* to PowderBed3D

tomflint22 committed:
Update Readme.md

updated readme in develop to cite softwareX paper

tomflint22 committed:
Merge branch 'master' into develop

d1ma4ek commented:
Hello. When installing the solver , it outputs
Non-miscible incompressible two-phase mixture.H
and at the beginning other errors in keLnIncludeAll: running wmakeLnInclude in dependent libraries:
wmakeLnInclude error: base directory /opt/openfoam9/src/transportModels/twoPhaseMixture/ does not exist

d1ma4ek commented:
Thanks for the reply. I could not download the package with openfoam 6 from the link ....
blueCFD-Core 2017 the terminal
gave another error in opening files and could not open file CUMULUS.H fatal error: MULES.H: No such file or directory
Where can I find openfoam 6?

d1ma4ek commented:
I will look forward to the update, but in the meantime I will look for other ways to solve this problem.

tomflint22 commented:
https://openfoam.org/version/6/

d1ma4ek commented:
it seems that due to inattention, I downloaded the patch, and not the directory itself. I also wanted to ask how you can set the arc pressure, according to the heating principle?

d1ma4ek commented:
> https://openfoam.org/version/6/

Hi Tom. Unfortunately, it is not possible to install opeanfoam6, for example, according to the link that you sent, it is no longer possible to download via the sudo apt-get -y install openfoam6 command.
I have tried installing tgz packages. In the same way, there are errors, especially if you do it on ubuntu from under windows.

I'm really looking forward to the new release for the new platform.
Maybe there is some task that I could complete and help you for the release?

tomflint22 commented:
Can you install openfoam10?
We have a laser ray tracing solver that compiles against openfoam10 if that is of interest while I find time to update beamWeldFoam 

https://github.com/micmog/LaserbeamFoam

d1ma4ek commented:
Thanks for the hint. Yesterday openfoam 10 worked. But I will reinstall Ubuntu one more time and install openfoam10.
Reading your article, I did not find a significant difference between arc heating and laser heating, it turns out that only the settings of the heat source differ?
I'm researching the VP-PAW process, and I wanted to ask how to set pressure to the boundary?

I had an idea to write a wrapper in C# for easy editing of initial parameters. But first you need to solve the problem with the installation.

tomflint22 commented:
so you can use the beamweldfoam solver to apply a surface Gaussian distribution. In the arc-case we just apply a surface gaussian representative of an arc welding case. Of course this neglects the Lorentz force and the joule heating. If you wanted to include that it is a lot more work. We have done this but the solver is not currently open-source.

You can read this article if you are interested though:

https://www.nature.com/articles/s41598-021-97177-8

d1ma4ek commented:
@tomflint22, Hello Tom. I would like to thank your team and you in particular for giving me the opportunity to use your work.
I did not want to, but I had to install Windows and Ubuntu 18.04 in parallel, now your code is running. To install openfoam6 I used this guide https://openfoamwiki.net/index.php/Installation/Linux/OpenFOAM-6/Ubuntu the old one will not work.

Arc welding actually has many nuances. But with plasma welding, this can be neglected, since the arc stabilizes. I'm interested in being able to apply pressure to a surface. Is it hard to implement in your code? The principle is probably the same as the heat source, only XY moments are needed.

Thanks for the article, I undertake to make the maximum number of references to your work.

Well, as I said, I will work on the C # interface for the convenience of settings. I am installing OpeanFoam for the first time and there is a lot of work ahead.

tomflint22 commented:
Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible.

d1ma4ek commented:
> Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible.

Hi Tom. Thanks for the info.
For the last 4 months I have been actively testing your solver.
I get really good results with heat input and describing the dimensions of the weld pool.
But there is a significant problem, in the 3D calculation the weld pool does not sag, or it sags much less.
I've already tried every option, surface tension, etc.

Here is one example...
May I contact you by mail to discuss?

tomflint22 commented:
Hi,

Yes sure. I’ve not seen these issues myself but I’m sure we can get to the bottom of it together.

***@***.***

I’m pretty busy but I’ll do my best to reply as soon as I can. It would help if you can make a Dropbox folder and upload one of these 3d cases there so I can take a look.

All the best,

Tom

Sent from my iPhone

On 22 Apr 2023, at 18:45, d1ma4ek ***@***.***> wrote:

﻿

Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible.

Hi Tom. Thanks for the info.
For the last 4 months I have been actively testing your solver.
I get really good results with heat input and describing the dimensions of the weld pool.
But there is a significant problem, in the 3D calculation the weld pool does not sag, or it sags much less.
I've already tried every option, surface tension, etc.

Here is one example...
May I contact you by mail to discuss?

[Снимок экрана от 2023-04-21 00-31-33]<https://user-images.githubusercontent.com/118846695/233799012-f0f6cf5d-23e2-4e9b-8a64-f572b94d8459.png>

—
Reply to this email directly, view it on GitHub<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1518715335>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2I6SDC4Y2D6IIWFRKDTXCQKKBANCNFSM5W4QXLEA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

d1ma4ek commented:
> Hi, Yes sure. I’ve not seen these issues myself but I’m sure we can get to the bottom of it together. ***@***.*** I’m pretty busy but I’ll do my best to reply as soon as I can. It would help if you can make a Dropbox folder and upload one of these 3d cases there so I can take a look. All the best, Tom
> […](#)
> Sent from my iPhone On 22 Apr 2023, at 18:45, d1ma4ek ***@***.***> wrote: ﻿ Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible. Hi Tom. Thanks for the info. For the last 4 months I have been actively testing your solver. I get really good results with heat input and describing the dimensions of the weld pool. But there is a significant problem, in the 3D calculation the weld pool does not sag, or it sags much less. I've already tried every option, surface tension, etc. Here is one example... May I contact you by mail to discuss? [Снимок экрана от 2023-04-21 00-31-33]<https://user-images.githubusercontent.com/118846695/233799012-f0f6cf5d-23e2-4e9b-8a64-f572b94d8459.png> — Reply to this email directly, view it on GitHub<[#6 (comment)](https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1518715335)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2I6SDC4Y2D6IIWFRKDTXCQKKBANCNFSM5W4QXLEA>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Hi Tom. 

Working with your example, 3D laser welding, I discovered the following.
1. Boundary conditions are chosen for the walls as a constant temperature of 300 K. This may affect the thermal design.

I did an experiment, TIG welding on an aluminum plate. I welded points with different welding times and got different diameters. Using the point diameter output parameters found the heat source parameters and adjusted them in your model.
Your model described the experiment well if only the heat input is taken into account.

I encountered the following problem, by emitting the gas kinetic pressure of the PAW welding jet, the deepening of the surface of the weld pool is good.
However, there is absolutely no sagging of the weld pool, or much less than the existing one.
This does not depend on the pressure of the gas stream jet, which will be directed to the surface.

I changed many parameters, density (aluminum), viscosity, surface tension from 0.6 to 0.8 N/m and surface tension gradient from 0.5*10-4 to 2.5*10-4. The result is either a hole or lack of proper sag.
The hole is the right solution for keyhole welding. However, if there is no hole, the sag is insufficient.

Please look at my model settings, maybe you can tell me some additional nuances with the model settings.
Welding takes place in argon, metal-aluminum, thickness 3 mm.
 
Download link
https://www.dropbox.com/scl/fo/kdytfpbdz5e36nao9neve/h?dl=0&rlkey=jmm33168esd5mp9azuyoi1fo1
 
https://drive.google.com/drive/folders/1br4SSCoU8qqXX_fiM2-GRf62aWqqdmnY?usp=share_link

tomflint22 commented:
Hi,

I had a look at your case set up.

I have a few pointers, but as I mentioned in a previous comment to you. The beamweldfoam solver does not contain the Lorentz force, or the 'arc-pressure' source.

You can read this paper for details on this, but we have not released this magnetohydrodynamics code yet:

https://www.nature.com/articles/s41598-021-97177-8

Dr Tom Flint, MPhys, MInstP

DSTL Research Fellow in Alloys for Extreme Environments

The University of Manchester, Department of Materials
Henry Royce Institute
Royce Hub Building
The University of Manchester
Manchester
M13 9PL
Mobile: +44 (0) 7732974085
e-mail : ***@***.***
Research Gate : https://www.researchgate.net/profile/Thomas-Flint
GitHub : https://github.com/tomflint22
________________________________
From: d1ma4ek ***@***.***>
Sent: 23 April 2023 20:00
To: tomflint22/beamWeldFoam ***@***.***>
Cc: Thomas Flint ***@***.***>; Mention ***@***.***>
Subject: Re: [tomflint22/beamWeldFoam] Powder bed fusion exampled and code clean up and readability edits (PR #6)


Hi, Yes sure. I’ve not seen these issues myself but I’m sure we can get to the bottom of it together. @.*** I’m pretty busy but I’ll do my best to reply as soon as I can. It would help if you can make a Dropbox folder and upload one of these 3d cases there so I can take a look. All the best, Tom
…
Sent from my iPhone On 22 Apr 2023, at 18:45, d1ma4ek @.> wrote: ﻿ Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible. Hi Tom. Thanks for the info. For the last 4 months I have been actively testing your solver. I get really good results with heat input and describing the dimensions of the weld pool. But there is a significant problem, in the 3D calculation the weld pool does not sag, or it sags much less. I've already tried every option, surface tension, etc. Here is one example... May I contact you by mail to discuss? [Снимок экрана от 2023-04-21 00-31-33]https://user-images.githubusercontent.com/118846695/233799012-f0f6cf5d-23e2-4e9b-8a64-f572b94d8459.png — Reply to this email directly, view it on GitHub<#6 (comment)<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1518715335>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AEGC2I6SDC4Y2D6IIWFRKDTXCQKKBANCNFSM5W4QXLEA. You are receiving this because you were mentioned.Message ID: @.>

Hi Tom.

Working with your example, 3D laser welding, I discovered the following.

  1.  Boundary conditions are chosen for the walls as a constant temperature of 300 K. This may affect the thermal design.

I did an experiment, TIG welding on an aluminum plate. I welded points with different welding times and got different diameters. Using the point diameter output parameters found the heat source parameters and adjusted them in your model.
Your model described the experiment well if only the heat input is taken into account.

I encountered the following problem, by emitting the gas kinetic pressure of the PAW welding jet, the deepening of the surface of the weld pool is good.
However, there is absolutely no sagging of the weld pool, or much less than the existing one.
This does not depend on the pressure of the gas stream jet, which will be directed to the surface.

I changed many parameters, density (aluminum), viscosity, surface tension from 0.6 to 0.8 N/m and surface tension gradient from 0.510-4 to 2.510-4. The result is either a hole or lack of proper sag.
The hole is the right solution for keyhole welding. However, if there is no hole, the sag is insufficient.

Please look at my model settings, maybe you can tell me some additional nuances with the model settings.
Welding takes place in argon, metal-aluminum, thickness 3 mm.

Download link
https://www.dropbox.com/scl/fo/kdytfpbdz5e36nao9neve/h?dl=0&rlkey=jmm33168esd5mp9azuyoi1fo1

https://drive.google.com/drive/folders/1br4SSCoU8qqXX_fiM2-GRf62aWqqdmnY?usp=share_link

—
Reply to this email directly, view it on GitHub<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1519133302>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2IYYURNBL3FUUPRCRVLXCV34HANCNFSM5W4QXLEA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

tomflint22 commented:
Sorry, I didnt mean to send that reply yet.


https://www.nature.com/articles/s41598-021-97177-8

I can see that you have tried to force a pressure gradient at the top surface. I'm not familiar with the fixedcoded BC, so I'm not sure how this works, but it certainly wont induce the Lorentz force in the weld-pool, it may give you an additional downward force but I'm guessing it's not doing what you expect.

I also see that your grading your mesh. You need to be careful with the solver as strictly speaking the heat needs to be applied in cells that are cubes for the heat deposition calculation to be correct.

Are you using the OpenFoam 10 version of the code? I think there are some redundant files on that brranch that are confusing things. I'llk see if I can clean it up now.

Dr Tom Flint, MPhys, MInstP

DSTL Research Fellow in Alloys for Extreme Environments

The University of Manchester, Department of Materials
Henry Royce Institute
Royce Hub Building
The University of Manchester
Manchester
M13 9PL
Mobile: +44 (0) 7732974085
e-mail : ***@***.***
Research Gate : https://www.researchgate.net/profile/Thomas-Flint
GitHub : https://github.com/tomflint22
________________________________
From: Thomas Flint ***@***.***>
Sent: 24 April 2023 14:29
To: tomflint22/beamWeldFoam ***@***.***>; tomflint22/beamWeldFoam ***@***.***>
Cc: Mention ***@***.***>
Subject: Re: [tomflint22/beamWeldFoam] Powder bed fusion exampled and code clean up and readability edits (PR #6)

Hi,

I had a look at your case set up.

I have a few pointers, but as I mentioned in a previous comment to you. The beamweldfoam solver does not contain the Lorentz force, or the 'arc-pressure' source.

You can read this paper for details on this, but we have not released this magnetohydrodynamics code yet:

https://www.nature.com/articles/s41598-021-97177-8

Dr Tom Flint, MPhys, MInstP

DSTL Research Fellow in Alloys for Extreme Environments

The University of Manchester, Department of Materials
Henry Royce Institute
Royce Hub Building
The University of Manchester
Manchester
M13 9PL
Mobile: +44 (0) 7732974085
e-mail : ***@***.***
Research Gate : https://www.researchgate.net/profile/Thomas-Flint
GitHub : https://github.com/tomflint22
________________________________
From: d1ma4ek ***@***.***>
Sent: 23 April 2023 20:00
To: tomflint22/beamWeldFoam ***@***.***>
Cc: Thomas Flint ***@***.***>; Mention ***@***.***>
Subject: Re: [tomflint22/beamWeldFoam] Powder bed fusion exampled and code clean up and readability edits (PR #6)


Hi, Yes sure. I’ve not seen these issues myself but I’m sure we can get to the bottom of it together. @.*** I’m pretty busy but I’ll do my best to reply as soon as I can. It would help if you can make a Dropbox folder and upload one of these 3d cases there so I can take a look. All the best, Tom
…
Sent from my iPhone On 22 Apr 2023, at 18:45, d1ma4ek @.> wrote: ﻿ Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible. Hi Tom. Thanks for the info. For the last 4 months I have been actively testing your solver. I get really good results with heat input and describing the dimensions of the weld pool. But there is a significant problem, in the 3D calculation the weld pool does not sag, or it sags much less. I've already tried every option, surface tension, etc. Here is one example... May I contact you by mail to discuss? [Снимок экрана от 2023-04-21 00-31-33]https://user-images.githubusercontent.com/118846695/233799012-f0f6cf5d-23e2-4e9b-8a64-f572b94d8459.png — Reply to this email directly, view it on GitHub<#6 (comment)<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1518715335>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AEGC2I6SDC4Y2D6IIWFRKDTXCQKKBANCNFSM5W4QXLEA. You are receiving this because you were mentioned.Message ID: @.>

Hi Tom.

Working with your example, 3D laser welding, I discovered the following.

  1.  Boundary conditions are chosen for the walls as a constant temperature of 300 K. This may affect the thermal design.

I did an experiment, TIG welding on an aluminum plate. I welded points with different welding times and got different diameters. Using the point diameter output parameters found the heat source parameters and adjusted them in your model.
Your model described the experiment well if only the heat input is taken into account.

I encountered the following problem, by emitting the gas kinetic pressure of the PAW welding jet, the deepening of the surface of the weld pool is good.
However, there is absolutely no sagging of the weld pool, or much less than the existing one.
This does not depend on the pressure of the gas stream jet, which will be directed to the surface.

I changed many parameters, density (aluminum), viscosity, surface tension from 0.6 to 0.8 N/m and surface tension gradient from 0.510-4 to 2.510-4. The result is either a hole or lack of proper sag.
The hole is the right solution for keyhole welding. However, if there is no hole, the sag is insufficient.

Please look at my model settings, maybe you can tell me some additional nuances with the model settings.
Welding takes place in argon, metal-aluminum, thickness 3 mm.

Download link
https://www.dropbox.com/scl/fo/kdytfpbdz5e36nao9neve/h?dl=0&rlkey=jmm33168esd5mp9azuyoi1fo1

https://drive.google.com/drive/folders/1br4SSCoU8qqXX_fiM2-GRf62aWqqdmnY?usp=share_link

—
Reply to this email directly, view it on GitHub<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1519133302>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2IYYURNBL3FUUPRCRVLXCV34HANCNFSM5W4QXLEA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

d1ma4ek commented:
> Sorry, I didnt mean to send that reply yet. https://www.nature.com/articles/s41598-021-97177-8 I can see that you have tried to force a pressure gradient at the top surface. I'm not familiar with the fixedcoded BC, so I'm not sure how this works, but it certainly wont induce the Lorentz force in the weld-pool, it may give you an additional downward force but I'm guessing it's not doing what you expect. I also see that your grading your mesh. You need to be careful with the solver as strictly speaking the heat needs to be applied in cells that are cubes for the heat deposition calculation to be correct. Are you using the OpenFoam 10 version of the code? I think there are some redundant files on that brranch that are confusing things. I'llk see if I can clean it up now. Dr Tom Flint, MPhys, MInstP DSTL Research Fellow in Alloys for Extreme Environments The University of Manchester, Department of Materials Henry Royce Institute Royce Hub Building The University of Manchester Manchester M13 9PL Mobile: +44 (0) 7732974085 e-mail : ***@***.*** Research Gate : https://www.researchgate.net/profile/Thomas-Flint GitHub : https://github.com/tomflint22
> […](#)
> ________________________________ From: Thomas Flint ***@***.***> Sent: 24 April 2023 14:29 To: tomflint22/beamWeldFoam ***@***.***>; tomflint22/beamWeldFoam ***@***.***> Cc: Mention ***@***.***> Subject: Re: [tomflint22/beamWeldFoam] Powder bed fusion exampled and code clean up and readability edits (PR #6) Hi, I had a look at your case set up. I have a few pointers, but as I mentioned in a previous comment to you. The beamweldfoam solver does not contain the Lorentz force, or the 'arc-pressure' source. You can read this paper for details on this, but we have not released this magnetohydrodynamics code yet: https://www.nature.com/articles/s41598-021-97177-8 Dr Tom Flint, MPhys, MInstP DSTL Research Fellow in Alloys for Extreme Environments The University of Manchester, Department of Materials Henry Royce Institute Royce Hub Building The University of Manchester Manchester M13 9PL Mobile: +44 (0) 7732974085 e-mail : ***@***.*** Research Gate : https://www.researchgate.net/profile/Thomas-Flint GitHub : https://github.com/tomflint22
> ________________________________ From: d1ma4ek ***@***.***> Sent: 23 April 2023 20:00 To: tomflint22/beamWeldFoam ***@***.***> Cc: Thomas Flint ***@***.***>; Mention ***@***.***> Subject: Re: [tomflint22/beamWeldFoam] Powder bed fusion exampled and code clean up and readability edits (PR #6) Hi, Yes sure. I’ve not seen these issues myself but I’m sure we can get to the bottom of it together. @.*** I’m pretty busy but I’ll do my best to reply as soon as I can. It would help if you can make a Dropbox folder and upload one of these 3d cases there so I can take a look. All the best, Tom … Sent from my iPhone On 22 Apr 2023, at 18:45, d1ma4ek @.> wrote: ﻿ Hi, there is now an OpenFoam10 version of beamweldfoam, on a separate branch. I hope this helps. Thanks for your nice words. We just want to make as many of our tools available as possible. Hi Tom. Thanks for the info. For the last 4 months I have been actively testing your solver. I get really good results with heat input and describing the dimensions of the weld pool. But there is a significant problem, in the 3D calculation the weld pool does not sag, or it sags much less. I've already tried every option, surface tension, etc. Here is one example... May I contact you by mail to discuss? [Снимок экрана от 2023-04-21 00-31-33]https://user-images.githubusercontent.com/118846695/233799012-f0f6cf5d-23e2-4e9b-8a64-f572b94d8459.png — Reply to this email directly, view it on GitHub<#6 (comment)<[#6 (comment)](https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1518715335)>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/AEGC2I6SDC4Y2D6IIWFRKDTXCQKKBANCNFSM5W4QXLEA. You are receiving this because you were mentioned.Message ID: @.> Hi Tom. Working with your example, 3D laser welding, I discovered the following. 1. Boundary conditions are chosen for the walls as a constant temperature of 300 K. This may affect the thermal design. I did an experiment, TIG welding on an aluminum plate. I welded points with different welding times and got different diameters. Using the point diameter output parameters found the heat source parameters and adjusted them in your model. Your model described the experiment well if only the heat input is taken into account. I encountered the following problem, by emitting the gas kinetic pressure of the PAW welding jet, the deepening of the surface of the weld pool is good. However, there is absolutely no sagging of the weld pool, or much less than the existing one. This does not depend on the pressure of the gas stream jet, which will be directed to the surface. I changed many parameters, density (aluminum), viscosity, surface tension from 0.6 to 0.8 N/m and surface tension gradient from 0.510-4 to 2.510-4. The result is either a hole or lack of proper sag. The hole is the right solution for keyhole welding. However, if there is no hole, the sag is insufficient. Please look at my model settings, maybe you can tell me some additional nuances with the model settings. Welding takes place in argon, metal-aluminum, thickness 3 mm. Download link https://www.dropbox.com/scl/fo/kdytfpbdz5e36nao9neve/h?dl=0&rlkey=jmm33168esd5mp9azuyoi1fo1 https://drive.google.com/drive/folders/1br4SSCoU8qqXX_fiM2-GRf62aWqqdmnY?usp=share_link — Reply to this email directly, view it on GitHub<[#6 (comment)](https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1519133302)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2IYYURNBL3FUUPRCRVLXCV34HANCNFSM5W4QXLEA>. You are receiving this because you were mentioned.Message ID: ***@***.***>

Thanks, Tom. Yes, I carefully read the article, and saw that need to use the cube.
I also understand that the time step is too large, as is the setting of the Courant number. I did this deliberately, since a gradual increase led to the same result.

The question is of interest: what parameter can affect the sagging of the liquid part of the metal. Not taking into account the Lorentz force, the liquid part should tend downwards, but this does not happen at all, even at a surface tension of 0.6 N.m. And when I apply pressure to the surface, it pulls in, not sags.
I also realized that some files in the constant folder are not responsible for the settings of the solid and liquid phase.

d1ma4ek commented:
Thanks, Tom. I hope you give me some guesses that I could use to check.

d1ma4ek commented:
Tom, I am guided by this work https://iopscience.iop.org/article/10.1088/0022-3727/32/22/312/pdf.
For example, after welding, the metal starts to sink down, why doesn't this happen in our case? The Lorentz force is no longer involved after the arc break.

There are certainly more relevant works, but I like the principle of description.

tomflint22 commented:
When you say the ’sagging’, do you mean the depression of the top surface?

As I mentioned, this depression of the top surface is due to the Lorentz force in arc welding, and is not present in beamweldfoam. What you have done to force a pressure gradient looks to me like it could work, bu I have no expertise I this kind of forcing of the pressure.

Dr Tom Flint, MPhys, MInstP

DSTL Research Fellow in Alloys for Extreme Environments

The University of Manchester, Department of Materials
Henry Royce Institute
Royce Hub Building
The University of Manchester
Manchester
M13 9PL
Mobile: +44 (0) 7732974085
e-mail : ***@***.******@***.***>
Research Gate : https://www.researchgate.net/profile/Thomas-Flint
GitHub : https://github.com/tomflint22

On 24 Apr 2023, at 20:15, d1ma4ek ***@***.******@***.***>> wrote:



Tom, I am guided by this work https://iopscience.iop.org/article/10.1088/0022-3727/32/22/312/pdf.
For example, after welding, the metal starts to sink down, why doesn't this happen in our case? The Lorentz force is no longer involved after the arc break.

There are certainly more relevant works, but I like the principle of description.

—
Reply to this email directly, view it on GitHub<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1520697266>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2I4B5KRY3S24LDSTPJDXC3GNJANCNFSM5W4QXLEA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

tomflint22 commented:
![t1_sag](https://user-images.githubusercontent.com/17575203/234269270-bc1041d3-15ad-49db-8f09-3366ab9b4ec8.png)
![t2_sag](https://user-images.githubusercontent.com/17575203/234269274-99379db6-abde-4706-a2f5-8c0448edef42.png)


Here is an example of a molten pool depressing/sagging due to gravity in beamweldfoam. But as I say, this is not due to the Lorentz force as this physics is not included in beamweldfoam.

I hope this helps. I just set a cylindrical region in the plate to be above the liquidus temperature here.

d1ma4ek commented:
Thanks, Tom. 'sagging' - I meant the bottom of the weld pool.
For some reason, in my calculation, the metal does not go down, but rather rises. Maybe it has something to do with pressure from above.

![12](https://user-images.githubusercontent.com/118846695/234286205-fcb5c2a6-e6dd-49f7-9569-cacf52e56093.png)

So I asked, maybe I configured something wrong?
Can you tell me the recommended grid and time step settings?

tomflint22 commented:
OK,

I understand now. It could be that your Marangoni, or dsigma/dT value is too large in magnitude maybe. Or maybe the buoyancy force is too great because the thermal expansion in your example - beta in the constant/thermophysicalproperties.metal - is too large. There are factors that may affect the flow and cause your melt pool to do weird things vs the experiment. 

How long are you running the simulation for? It could also be that you haven't run the simulation long enough to see this sagging. Why don't you try getting rid of that pressure boundary condition for now, and turn off the heat source, and just set the central region to be above the liquids temperature - that way you should be able to increase the time-step size and decrease the mesh resolution and run the simulation for a few ms, to see that you get sagging under these conditions. If you don't then I'd suggest that either your Marangoni coefficient or thermal expansion may be too large.

What is the value you are using for the viscosity of the metal phase? The solver expects kinematic viscosity here.

d1ma4ek commented:
I would not say that this is not a correct solution, since in the experiment a keyhole is obtained during solidification (crater).
I use the following parameters for surface tension and surface tension gradient.
aluminum
phase1
{
     transportModel Newtonian;
     nu 0.0008;
     rho 2300;
     cp 1100;
     cpsolid 896.0;
     kappa 100.0;
kappasolid 169.0;
Tsolidus 931;
Tliquidus 933;
     LatentHeat 396666;
     beta 2.35e-5;
}
argon
phase2
{
     transportModel Newtonian;
     nu 1.48e-05;
     rho 1.78;
     cp 500;
cpsolid 500;
     kappa 0.016;
kappasolid 0.016;
     Tsolidus 1.0;
     Tliquidus 10.0;
     latent heat 1;
     beta 3.76e-3;
}

sigma 0.7;//1.5;
dsigmadT -0.5e-4;

I can only say that I have already done calculations without pressure and my experiment on the diameter of the deposit was accurately described by your solver. With the settings I attached earlier.

Correct me if I did something wrong here.

I'll take your advice and go back to a simpler option, but what are the initial parameters for the time step and grid size to be set?

tomflint22 commented:
Hi,

Yes I was just suggesting this simplified model so that you can track down your problem.

I think I can see though that you are inputting the dynamic viscosity for aluminium, the kinematic viscosity is this value divided by the material density. So I would expect your nu value for aluminium to be of the order 1e-6 not 1e-3. You’re simulating something that is about 1000 times more viscous than aluminium, but this is an easy mistake to make.

All the best,

Tom

Sent from my iPhone

On 25 Apr 2023, at 14:36, d1ma4ek ***@***.***> wrote:

﻿

I would not say that this is not a correct solution, since in the experiment a keyhole is obtained during solidification (crater).
I use the following parameters for surface tension and surface tension gradient.
aluminum
phase1
{
transportModel Newtonian;
nu 0.0008;
rho 2300;
cp 1100;
cpsolid 896.0;
kappa 100.0;
kappasolid 169.0;
Tsolidus 931;
Tliquidus 933;
LatentHeat 396666;
beta 2.35e-5;
}
argon
phase2
{
transportModel Newtonian;
nu 1.48e-05;
rho 1.78;
cp 500;
cpsolid 500;
kappa 0.016;
kappasolid 0.016;
Tsolidus 1.0;
Tliquidus 10.0;
latent heat 1;
beta 3.76e-3;
}

sigma 0.7;//1.5;
dsigmadT -0.5e-4;

I can only say that I have already done calculations without pressure and my experiment on the diameter of the deposit was accurately described by your solver. With the settings I attached earlier.

Correct me if I did something wrong here.

—
Reply to this email directly, view it on GitHub<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1521803700>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2I7F47JVYAK3BWNPX4DXC7HMXANCNFSM5W4QXLEA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

d1ma4ek commented:
Thanks a lot, Tom.
I really did not pay attention to it, I will check the solution.
Thank you thank you thank you.

tomflint22 commented:
No problem, I'm happy to help. I actually made the same mistake once and spent ages wondering why my flow looked weird.

Let me know how you get on. I hope that pressure boundary condition works for you.

All the best,

Tom

d1ma4ek commented:
Oh yes, Tom, I'm very familiar with that. I'll be sure to let you know my results.

d1ma4ek commented:
Hi Tom. For the last month I have been trying to achieve the effect of lowering liquid metal. Pressure gradient boundary condition, not used. Viscosity did not produce a significant result.
1. Why do 2D and 3D differ significantly in thermal calculation, how can they be made equivalent? For example 3D 1000 W = 2D .... W?
I used a cube grid, 1mm/7 with 10e-05 time steps.
I apologize for the possibly inappropriate questions.
But the thermal calculation at the same power in 2D and 3D does not converge.
The lowering of the weld pool also does not occur. Maybe I'm doing something wrong, please help. Thank you.

tomflint22 commented:
Hi,

The code is designed for 3D simulations really. Openfoam can do 2D simulations but in our case a 2D simulation is a cross section through the surface Gaussian distribution and so you arent capturing all the heat. Also the conversion from the surface flux to volumetric flux assumes the cells where the heat is applied are regular cuboids.

I'm sorry. I provided you with the answer to this 'lowering' of the weld pool already. This is not an issue with the code, it's an issue with your case set-up. I'm afraid I really dont have time to set your cases up for you. I'm really happy to help but I've given as much of my time as I can to this issue. Please see my previous responses. Last time your viscosity was 1000 times too large.

All the best,

Tom



Dr Tom Flint, MPhys, MInstP

DSTL Research Fellow in Alloys for Extreme Environments

The University of Manchester, Department of Materials
Henry Royce Institute
Royce Hub Building
The University of Manchester
Manchester
M13 9PL
Mobile: +44 (0) 7732974085
e-mail : ***@***.***
Research Gate : https://www.researchgate.net/profile/Thomas-Flint
GitHub : https://github.com/tomflint22
________________________________
From: d1ma4ek ***@***.***>
Sent: 25 May 2023 06:07
To: tomflint22/beamWeldFoam ***@***.***>
Cc: Thomas Flint ***@***.***>; Mention ***@***.***>
Subject: Re: [tomflint22/beamWeldFoam] Powder bed fusion exampled and code clean up and readability edits (PR #6)


Hi Tom. For the last month I have been trying to achieve the effect of lowering liquid metal. Pressure gradient boundary condition, not used. Viscosity did not produce a significant result.

  1.  Why do 2D and 3D differ significantly in thermal calculation, how can they be made equivalent? For example 3D 1000 W = 2D .... W?
I used a cube grid, 1mm/7 with 10e-05 time steps.
I apologize for the possibly inappropriate questions.
But the thermal calculation at the same power in 2D and 3D does not converge.
The lowering of the weld pool also does not occur. Maybe I'm doing something wrong, please help. Thank you.

—
Reply to this email directly, view it on GitHub<https://github.com/tomflint22/beamWeldFoam/pull/6#issuecomment-1562275199>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AEGC2I26ZFMYVWUYB3DZDLTXH3SHRANCNFSM5W4QXLEA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

tomflint22 commented:
Hi,

That looks correct to me to be honest. In 3D I think you just have more complex temperature gradients.

On 12 May 2024, at 18:13, Franz ***@***.***> wrote:



Hi, Tom. I found an oddity in the solver. When using the YZ (2D) solution, the Marangoni surface tension force(Magnitude) in Y coordinates has negative values for the lower surface.
2024-05-13.08-51-41.png (view on web) [github.com]<https://urldefense.com/v3/__https://github.com/tomflint22/beamWeldFoam/assets/118846695/4263e08f-e85d-48fc-8f4e-ca1a7f36df23__;!!PDiH4ENfjr2_Jw!DACcf4J-hB9loLzFZTuYNuWRIN3SBj2kGmqHltZTQeDO33Eg3J0jngYEnFCpy-I6Ehm_8_1v2j2qDBNh5vTiI9imZ_r1RJxuuw$>

If we consider the XYZ (3D) solution, then Marangoni(Magnitude) in the Y coordinate has a positive sign for the lower surface.
Such a phenomenon appears only in the presence of a temperature gradient. As you have shown above, there is no such phenomenon at a constant temperature. It also does not manifest itself if you set dy/dT = 0.1e-4 H/m^2*K, i.e. 10 times less.
2024-05-13.09-12-00.png (view on web) [github.com]<https://urldefense.com/v3/__https://github.com/tomflint22/beamWeldFoam/assets/118846695/b3a992da-5a43-449b-9313-18cc5d2b75b3__;!!PDiH4ENfjr2_Jw!DACcf4J-hB9loLzFZTuYNuWRIN3SBj2kGmqHltZTQeDO33Eg3J0jngYEnFCpy-I6Ehm_8_1v2j2qDBNh5vTiI9imZ_qkV2olwA$>

Analyzing the code and the original equations, I understand that the force is calculated in the solver as Marangoni_Constant*(gradT-nHatM*(nHatM & gradT))*mag(gradAlpha).
What could be the reason?
I have seen from many examples that I am setting the correct dy/dT as well as beta
sorry to disturb I've spent a year studying this process, and I haven't found a specific reason.

As for the thermal processes, they perfectly describe the experiment.

I also introduced an additional term of pressure to the surface, using the example of vapor pressure.

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.com/v3/__https://github.com/tomflint22/beamWeldFoam/pull/6*issuecomment-2106317903__;Iw!!PDiH4ENfjr2_Jw!DACcf4J-hB9loLzFZTuYNuWRIN3SBj2kGmqHltZTQeDO33Eg3J0jngYEnFCpy-I6Ehm_8_1v2j2qDBNh5vTiI9imZ_qnsyh5aQ$>, or unsubscribe [github.com]<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AEGC2IYIVNTGPCWA7D4J5HTZB6PKVAVCNFSM5W4QXLEKU5DIOJSWCZC7NNSXTN2JONZXKZKDN5WW2ZLOOQ5TEMJQGYZTCNZZGAZQ__;!!PDiH4ENfjr2_Jw!DACcf4J-hB9loLzFZTuYNuWRIN3SBj2kGmqHltZTQeDO33Eg3J0jngYEnFCpy-I6Ehm_8_1v2j2qDBNh5vTiI9imZ_of0LGd-g$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

d1ma4ek commented:
Hi, Tom. I continued working with your beamWeldFoam solver.
I know that you are working on another LaserbeamFoam.

However, I like the heat source better in this solver.

Therefore, I added, according to the principle of a heat source, a pressure source to the surface.

That's how it works.
![Снимок экрана от 2024-05-15 23-04-35](https://github.com/tomflint22/beamWeldFoam/assets/118846695/ed6b07a5-26db-47f2-9659-22cdee8f33db)
![Снимок экрана от 2024-05-15 23-05-34](https://github.com/tomflint22/beamWeldFoam/assets/118846695/de6d5db8-9155-44ce-be87-096f1661eff1)


However, I can't use it as long as there is a problem with the bottom surface.

I have a suspicion that the marangoni force in 3D is calculated differently.

I think it 's an option . Maybe this will help?
http://dx.doi.org/10.1016/j.matdes.2016.01.019

tomflint22 commented:
Hi, it's not calculated differently in 3D. It's probably that your boundary condition in T isn't quite correct, or you need to be further away from the boundary

d1ma4ek commented:
> Hi, it's not calculated differently in 3D. It's probably that your boundary condition in T isn't quite correct, or you need to be further away from the boundary

I want you to understand that the thermal solution fits perfectly.
The top surface works fine, however



The lower surface raises the weld pool too much, precisely because of the surface tension gradient.

This does not affect in any way whether there is arc pressure or not. You can test it yourself.

A physical experiment with a certain size of the welding bath gives it a downward movement. Everything is different in the model.

I really ask you to simulate with a gradient, and see what the reason may be in 3D. Because it works in 2D...

tomflint22 commented:
Yes I understand, but what I am saying is that based on the thermal solution the Marangoni force is being calculated correctly. It's just that this force dominates, so if your T boundary condition is not perfect, yo can get weird things happening. Is your set-ip the same as you sent previously? I suggest:
*modifying the domain so that your metal is further away from that bottom boundary
*tweaking that bottom T boundary condition
*maybe tweaking the alpha.metal boundary on the bottom to be the same as the top boundary (if they're not already)

tomflint22 commented:
> ![t1_sag](https://user-images.githubusercontent.com/17575203/234269270-bc1041d3-15ad-49db-8f09-3366ab9b4ec8.png) ![t2_sag](https://user-images.githubusercontent.com/17575203/234269274-99379db6-abde-4706-a2f5-8c0448edef42.png)
> 
> Here is an example of a molten pool depressing/sagging due to gravity in beamweldfoam. But as I say, this is not due to the Lorentz force as this physics is not included in beamweldfoam.
> 
> I hope this helps. I just set a cylindrical region in the plate to be above the liquidus temperature here.

We showed here that it won't be because of the Marangoni implementation explicitly, but maybe the heating in your simulation gives a stronger/different direction thermal gradient which will give the Marangoni force in the wrong direction. This is a boundary condition issue

d1ma4ek commented:
Okay, but you didn't provide the initial data, what value of the surface tension gradient did you set?
For example, if you take -0.123e3 H/m*K for aluminum, you get one result, if -0.123e4 H/m*K, you get a completely different result.

The second case leads to values 10 times smaller, and therefore the simulation works.
I use -0,123e3 H/m*K.

tomflint22 commented:
Exactly, that's how sensitive the simukation is to the thermal fields. 
Please try my suggestions then get back to me

d1ma4ek commented:
> Exactly, that's how sensitive the simukation is to the thermal fields. Please try my suggestions then get back to me

Tom, I'm sorry. Here is my gradient.
![Снимок экрана от 2024-05-16 00-27-26](https://github.com/tomflint22/beamWeldFoam/assets/118846695/45b56d38-00ac-45f6-8fd0-930a66fa7b74)
![Снимок экрана от 2024-05-16 00-27-59](https://github.com/tomflint22/beamWeldFoam/assets/118846695/26d3363b-361b-4026-9032-1f9256a03421)

Perhaps the fact is that in your version there is no gradient in height.
